packages/extension-driver-snowflake/src/lib/snowflakeDataSource.ts

```diff
@@ -100,6 +97,78 @@ export class SnowflakeDataSource extends DataSource<any, SnowflakeOptions> {
     }
   }
 
+  public override async export(exportOptions: ExportOptions): Promise<void> {
+    let { directory } = exportOptions;
+    const { profileName, sql } = exportOptions;
+
+    const { connection, pool } = await this.getConnection(profileName);
+    directory = /^\./.exec(directory) ? directory : `./${directory}`;
+
+    // throw error if filepath dir not exist
+    if (!fs.existsSync(directory)) {
+      throw new InternalError(`Directory ${directory} not exist`);
+    }
+
+    try {
+      const stage = '@~/vulcan_cache_stage';
+      const stageFilePath = `${stage}/`;
+      const builtSQL = this.getCopyToStageSQL(sql, stageFilePath);
+      // copy data to a named stage
+      const copyStatement = await this.getStatementFromSQL(
+        connection,
+        builtSQL,
+        []
+      );
+      const statementId = copyStatement.getStatementId();
+      const pattern = `.*${statementId}.*`; // UUID generated by snowflake
+      await this.waitStatement(copyStatement);
+      this.logger.debug('Copied data to internal stage: @~/vulcan_cache_stage');
+
+      // get parquet file from a named stage
+      const absoluteDir = path.resolve(directory);
+      const getStatement = await this.getStatementFromSQL(
+        connection,
+        `GET ${stage} file://${absoluteDir} pattern='${pattern}'`,
+        []
+      );
+      await this.waitStatement(getStatement);
+      this.logger.debug(`Exported parquet files to ${directory}`);
+
+      // remove parquet file from stage
+      await this.getStatementFromSQL(
+        connection,
+        `REMOVE ${stage} pattern='${pattern}'`,
+        []
+      );
+      this.logger.debug('Removed parquet files from stage');
+    } catch (err) {
+      this.logger.error(`Error when exporting parquet "${directory}": `, err);
+      throw err;
+    } finally {
+      this.logger.debug(`Release connection to ${profileName}`);
+      pool.release(connection);
+    }
+  }
+
+  // use protected for testing purpose
+  protected getCopyToStageSQL(sql: string, stageFilePath: string) {
+    return `COPY INTO ${stageFilePath} FROM (${sql}) FILE_FORMAT = (TYPE = 'parquet') HEADER=true INCLUDE_QUERY_ID=true MAX_FILE_SIZE=5368709120;`;
+  }
+
+  private async waitStatement(statement: snowflake.Statement) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return new Promise((resolve, reject) => {
+      let status = statement.getStatus();
+      while (status === 'fetching') {
+        status = statement.getStatus();
+        if (status === 'complete') {
+          resolve(true);
+        }
+      }
+      resolve(true);
```

Comment:
Seems that we do not get the `waitStatement` result to use, and here no `resolve(false)` case need, so we could only `resolve()` directly 😄 

Btw, The [snowflak-sdk](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/4783f0ca2d8f4d859cf4d9a1663818de993c5ef8/lib/connection/statement.js#L531) only has two status: `fetching` and `complete`, so maybe we could only check the while not equal to `'complete'` directly and if the status is `complete`, then it will leave the loop and `resolve()` directly:

```ts
return new Promise((resolve, reject) => {
      const status = statement.getStatus();
      while (status !== StatementStatus.Complete);
      resolve();
    });
```